### PR TITLE
test: Bump shellcheck version

### PR DESCRIPTION
**Pull request by hebasto**
The changelog for v0.7.2 is available [here](https://github.com/koalaman/shellcheck/blob/v0.7.2/CHANGELOG.md).

Only [SC2268](https://github.com/koalaman/shellcheck/wiki/SC2268) requires to update our code.

### CODE REVIEW

File: depends/config.site.in [hunk 1]
```diff
@@ -62,7 +62,7 @@ if test -z "$enable_zmq" && test -n "@no_zmq@"; then
   enable_zmq=no
 fi
 
-if test "x@host_os@" = xdarwin; then
+if test "@host_os@" = darwin; then
```
Comment (fanquake) [inline depends/config.site.in:65]: How is this related to this change

Comment (hebasto) [reply depends/config.site.in:65]: New [SC2268](https://github.com/koalaman/shellcheck/wiki/SC2268) added.

**Comment by kiminuo** [comment]
Concept ACK 

The [changelog](https://github.com/koalaman/shellcheck/blob/v0.7.2/CHANGELOG.md#v072---2021-04-19) shows minor useful changes.

**Review by jarolrod — COMMENTED** [review]
ACK  08f3dbb1b0cd5ca01d87e488a2fa905adf7df057

Patch looks correct and CI is green.

grabbed 0.7.2 and ran linter locally with no problem on linux.
